src/client/sound/s_mix.c

```diff
@@ -58,17 +60,31 @@ static _Bool S_SpatializeChannel(s_channel_t *ch) {
 
 	if (ch->play.flags & S_PLAY_POSITIONED) {
 		VectorCopy(ch->play.origin, org);
+		VectorClear(ch->velocity);
 	} else if (ch->play.flags & S_PLAY_ENTITY) {
 		const cl_entity_t *ent = &cl.entities[ch->play.entity];
+		
+		if (ent == cl.entity) {
+			VectorCopy(cl.frame.ps.pm_state.velocity, ch->velocity);
+		}
+		else {
+			VectorSubtract(ent->current.origin, ent->prev.origin, ch->velocity);
+		}
+
 		if (ent->current.solid == SOLID_BSP) {
 			const r_model_t *mod = cl.model_precache[ent->current.model1];
 			VectorLerp(mod->mins, mod->maxs, 0.5, center);
 			VectorAdd(center, ent->origin, org);
 		} else {
-			VectorCopy(ent->origin, org);
+			if (ent == cl.entity) {
+				VectorCopy(r_view.origin, org);
+			} else {
+				VectorCopy(ent->origin, org);
+			}
```

Comment:
This is how it worked in Mixer, but OpenAL requires this adjustment otherwise the sound will fade away as it's being made.